src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -0,0 +1,126 @@
+package org.mifos.connector.mpesa.camel.routes;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.json.JSONObject;
+import org.mifos.connector.mpesa.auth.AccessTokenStore;
+import org.mifos.connector.mpesa.dto.BuyGoodsPaymentRequestDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.ACCESS_TOKEN;
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.BUY_GOODS_REQUEST_BODY;
+import static org.mifos.connector.mpesa.safaricom.config.SafaricomProperties.MPESA_BUY_GOODS_TRANSACTION_TYPE;
+import static org.mifos.connector.mpesa.utility.SafaricomUtils.getPassword;
+
+
+@Component
+public class SafaricomRoutesBuilder extends RouteBuilder {
+
+    @Value("${mpesa.api.passKey}")
+    private String passKey;
+
+    @Value("${mpesa.api.lipana}")
+    private String buyGoodsBaseUrl;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AccessTokenStore accessTokenStore;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public void configure() {
+
+        /*
+           Use this endpoint for receiving the callback form safaricom mpesa endpoint
+         */
+        from("rest:POST:/buygoods/callback")
+                .id("buy-goods-callback")
+                .setBody(exchange -> {
+                    String body = exchange.getIn().getBody(String.class);
+                    JSONObject object = new JSONObject(body);
+                    return object.toString();
+                });
```

Comment:
What happens after this? 

There needs to be correlation between request and transaction id. After you receive the callback you need to publish a message to Zeebe stating the failure or success of transaction.

Similar to: https://github.com/openMF/ph-ee-connector-gsma-mm/blob/fef9c82a11642cdf1fba762a5d02d909740ea940/src/main/java/org/mifos/connector/gsma/transfer/TransferResponseProcessor.java#L48

Reply:
Zeebeworkflow is not designed until, so most probably it will got to publish message state, which Ankita is working on currently.

Reply:
I am not familiar with Zeebe workflow design, so I am reading the docs and deployment part. Also can we use [this](https://cawemo.com/share/cc998d9f-19ea-41c7-b72e-30ea3ff1a4fe) workflow which you shared earlier?

Reply:
I am not sure how to implement the `ConfirmTransaction` and `GetTransactionStatus` flow, because the confirmTransaction is a callback and we have to wait for some time and if in that time the callback is not received then we have to manually call the GetTransaction API and decide either it Is failed case or success case. So how to implement the timed task like this will be executed after n number of seconds after occurring of certain event.

Reply:
Resolving this. We have addressed this in BPMN during calls.